### ISSUE TITLE
Error message: "The database has not finished Ioading. The check for the presence of the file in the archive has been skipped"

### ISSUE THREAD

**Issue by Akseli-Ilmanen** [Mattk70/Chirpity-Electron#80]
Hi Matt!

When I try to upload an audio file, the GUI does not respond and sometimes I receive this error message "The database has not finished Ioading. The check for the presence of the file in the archive has been skipped" (see also screenshot). 

I wasn't sure what this meant and what I can do to fix this?

**Screenshots**
![image](https://github.com/Mattk70/Chirpity-Electron/assets/107996462/4f2bd802-048b-4442-be8a-3f56512fdd36)

Appreciate the help!

Best
Akseli

**Comment by Mattk70** [comment]
Hi Akseli,

I guess you mean "open a file"? There's no upload feature.

The message is there as it is possible to open a file before the database has finished loading - when you first start the application. Whenever you open a file, Chirpity checks the database to see if the file is known to it, and if it is, you can retrieve the results associated wtih that file. However, this should only happen one time if you fire up the app and then open a file very quickly - if you were to try again, the database should have loaded - it only takes a second or two.

It may be that some other issue has prevented the database from being loaded. I can see one issue in the error logs the app send me which suggests - so far as I can tell - an issue during the application's installation.

I suggest you uninstall the application (using add / remove programs in windows) and download / install a fresh copy of the application. Let me know if that resolves this issue?

**Comment by Akseli-Ilmanen** [comment]
Hi @Mattk70,

Thanks for the quick reply!

Yes, sorry, I meant "open file" not "upload".

So, I tried uninstalling and reinstalling it a couple of times. The first time after each installation, I am able to open a file (see pic below). However, if I then close Chirpity and open the program again, then I run into the "The database has not finished Ioading." error again. I also gave Chirpity a few minutes to run ("to load the database"), but that did not help.

Moreover, I noticed that my audio file and the associated spectrogram in Chirpity is squished into the 0-5kHz range, with the 5-10kHz range being nearly blank. See the "squishing" in the picture below. I guess this is not related to the database loading error but a different issue?

![image](https://github.com/Mattk70/Chirpity-Electron/assets/107996462/64334bee-bf30-45d7-bb5f-dce3274e772a)

Appreciate the help!

Best
Akseli

**Comment by Mattk70** [comment]
Hmm. 

I believe the problem you are having is due to this error:
`A dynamic link library (DLL) initialization routine failed `
This error gives further details about the tensorflow node bindings (you should be able to see these errors if you relaunch the application in debug mode - which you can enable in settings). It is most likely the root cause of other issues and errors.

This will be difficult for me to debug, as I cannot replicate the issue on either of the windows machines I have access to. 

However, an issue of this kind is mentioned here: https://github.com/tensorflow/tfjs/issues/3634

In that instance, the error was due to the CPU not supporting an instruction set used by Tensorflow. So - could you please send me a screenshot of the Help > Diagnostics panel?

Many thanks!

**Comment by Mattk70** [comment]
I have managed to replicate these exact errors by disabling AVX extensions on my computer BIOS. @Akseli-Ilmanen - I will close this issue, and add a note on minimum requirements (i.e. AVX support needed) to the download page.

Do let me know if your PC's CPU _does_ have AVX support (and it is not disabled) and I will investigate further.

**Comment by Akseli-Ilmanen** [comment]
Yeah, it is working now, thank you so much!

**Comment by Mattk70** [comment]
That’s great! Could you tell me - was the issue to do with AVX extensions? Were they disabled?

**Comment by Akseli-Ilmanen** [comment]
Sorry, my last message was a bit misleading. I recently build a PC and chirpity is working on this new PC, that's why I wrote "it's working".

On my old laptop (a Lenonov Intel(R) Celeron(R) N5100 @ 1.10GHz   1.11 GHz), I just looked for AVX in my BIOS and did not find anything.I hope this helps. 

For me personally, it will be perfectly good to work with Chirpity on the new PC and there is no need to resolve the AVX thing.
